### PR TITLE
Rename DefaultStateManager to MerkleStateManager

### PR DESCRIPTION
**Pull request by scorbajio**
The aim of this change is to rename the `DefaultStateManager` to `MerkleStateManager` as discussed in #3631 and to make the naming consistent throughout the monorepo.

### CODE REVIEW

File: packages/statemanager/test/statelessVerkleStateManager.spec.ts [hunk 1]
```diff
@@ -124,11 +124,27 @@ describe('StatelessVerkleStateManager: Kaustinen Verkle Block', () => {
     const address = createAddressFromString('0x6177843db3138ae69679a54b95cf345ed759450d')
     const stem = getVerkleStem(stateManager.verkleCrypto, address, 0n)
 
+    console.log(bytesToHex(stem))
```
Comment (gabrocheleau) [inline packages/statemanager/test/statelessVerkleStateManager.spec.ts:127]:
```suggestion
```

File: packages/statemanager/test/statelessVerkleStateManager.spec.ts [hunk 2]
```diff
@@ -124,11 +124,26 @@ describe('StatelessVerkleStateManager: Kaustinen Verkle Block', () => {
     const address = createAddressFromString('0x6177843db3138ae69679a54b95cf345ed759450d')
     const stem = getVerkleStem(stateManager.verkleCrypto, address, 0n)
 
+
     const basicDataKey = getVerkleKey(stem, VerkleLeafType.BasicData)
     const codeHashKey = getVerkleKey(stem, VerkleLeafType.CodeHash)
 
+    console.log('dbg100')
```
Comment (gabrocheleau) [inline packages/statemanager/test/statelessVerkleStateManager.spec.ts:131]: Lots of console logs that can be removed in this file

File: packages/statemanager/test/vmState.spec.ts [hunk 3]
```diff
@@ -47,7 +47,7 @@ describe('stateManager', () => {
     ]
 
     for (const [chain, expectedStateRoot] of chains) {
-      const stateManager = new DefaultStateManager({})
+      const stateManager = new MerkleStateManager({})
```
Comment (gabrocheleau) [inline packages/statemanager/test/vmState.spec.ts:50]: Is the empty object necessary?

**Review by gabrocheleau — APPROVED** [review]

**Comment by holgerd77** [comment]
Cool! 😎 

@gabrocheleau @scorbajio (others) so are you guys happy with this renaming as well? Just for my own confirmation (or not), didn’t want to simply push through, but personally (obviously) a fan and also think that I read the idea couple of times in the chat before.

**Comment by scorbajio** [comment]
> Cool! 😎
> 
> @gabrocheleau @scorbajio (others) so are you guys happy with this renaming as well? Just for my own confirmation (or not), didn’t want to simply push through, but personally (obviously) a fan and also think that I read the idea couple of times in the chat before.

Yes, this naming is more descriptive than just calling it "default" at this point.

**Comment by holgerd77** [comment]
Nice, thanks guys for the feedback! 🙂👍

**Pull request merged**